deployment/README.md

```diff
@@ -27,51 +100,27 @@ and then exposed for use in persistent environments like testnet/mainnet.
 contracts (like MCMS, LinkToken etc) which can be shared
 by products.
 
-/deployment/ccip
-- package name `ccipdeployment`
-- Files and tests per product deployment/configuration workflows
-- Tests can use deployment/memory for fast integration testing
-- TODO: System state representation is defined here, need to define
-  an interface to comply with for all products.
-
-/deployment/ccip/changeset
-- package name `changeset` imported as `ccipchangesets`
-- These function like scripts describing state transitions
-  you wish to apply to _persistent_ environments like testnet/mainnet
-- They should be go functions where the first argument is an
-  environment and the second argument is a config struct which can be unique to the 
-  changeset. The return value should be a `deployment.ChangesetOutput` and an error.
-
--  `do_something.go`
-    ```Go
-    func DoSomethingChangeSet(env deployment.Environment, c ccipdeployment.Config) (deployment.ChangesetOutput, error)
-    {
-        // Deploy contracts, generate MCMS proposals, generate
-        // job specs according to contracts etc.
-        return deployment.ChangesetOutput{}, nil
-    }
-    ```
-
--  `do_something_test.go`
-    ```Go
-    func TestDoSomething(t *testing.T)
-    {
-        // Set up memory env
-        // DoSomethingChangeSet function
-        // Take the artifacts from ChangeSet output
-        // Apply them to the memory env
-        // Send traffic, run assertions etc.
-    }
-    ```
-- Changesets are exposed and applied via a different repo. 
-
-/deployment/llo
-- package name `llodeployment`
-- Similar to /deploymet/ccip, these are product-specific deployment/configuration workflows
-- Tests can use deployment/memory for fast integration testing
-
-/deployment/llo/changeset
-- package name `changeset` imported as `llochangesets`
-- Similar to deployment/ccip/changesets
-- These function like scripts describing state transitions
-  you wish to apply to _persistent_ environments like testnet/mainnet
+/deployment/<product>
+- package name `<product>deployment`
+- Internal building blocks for changesets  
+- TODO: can we make this `internal`?
```

Comment:
If changesets are the public API of the deployments package then probably yes. Would also mean that we don't break things w/ out knowing for importers of this package.

Reply:
I think that's certainly the goal of moving towards internal use. We want to isolate points of contact to truly public information, shared types/apis/etc. and let implementation details stay isolated.

There may be some common code you want to be available for use in integration_tests so not everything would be internal, but it would force an evaluation of APIs, publicity, and that explicit determination is healthy.